usaspending_api/settings.py

```diff
@@ -279,17 +279,26 @@ def _configure_database_connection(environment_variable):
 # (which is "DATABASE_URL" by default). Generally speaking, DB_SOURCE is used to support server
 # environments that support the API/website and docker-compose local setup whereas DATABASE_URL
 # is used for development and operational environments (Jenkins primarily). If DB_SOURCE is provided,
-# then DB_R1 (read replica) must also be provided.
+# then DB_R1 (read replica) and DOWNLOAD_DATABASE_URL (primary or read replica depending on environment)
+# must also be provided.
 if os.environ.get("DB_SOURCE"):
     if not os.environ.get("DB_R1"):
         raise EnvironmentError("DB_SOURCE environment variable defined without DB_R1")
+    if not os.environ.get("DOWNLOAD_DATABASE_URL"):
+        raise EnvironmentError("DB_SOURCE environment variable defined without DOWNLOAD_DATABASE_URL")
     DATABASES = {
         DEFAULT_DB_ALIAS: _configure_database_connection("DB_SOURCE"),
         "db_r1": _configure_database_connection("DB_R1"),
+        "db_download": _configure_database_connection("DOWNLOAD_DATABASE_URL"),
```

Comment:
This generally looks okay, but it'd be good to talk through the need for why this all the sudden has to be in settings and we got by without it before. i.e. what changed? Is it that the python code running in BD needs to distinguish databases?

Also, as it's written here, will all envs and env-component-configs now need to provide the `DOWNLOAD_DATABASE_URL`? even deployed API components?

Reply:
Added this in order to use `.using()` in the Django ORM to specify which DB to hit for the Queryset. Probably better to conditionally add this if `DOWNLOAD_DATABASE_URL` is provided. 

It isn't that the value is all of sudden needed in `settings.py` but that it is needed in the `DATABASES` setting. Currently `DOWNLOAD_DATABASE_URL` is used directly as a connection string in the PSQL command.